src/azure-cli-core/azure/cli/core/extension/__init__.py

```diff
@@ -374,3 +374,9 @@ def is_preview_from_semantic_version(version):
     from packaging.version import parse
     parsed_version = parse(version)
     return bool(parsed_version.pre and parsed_version.pre[0] in ["a", "b"])
+
+
+def is_stable_from_metadata(item):
+    return not (item["metadata"].get(EXT_METADATA_ISPREVIEW, False) or
+                item["metadata"].get(EXT_METADATA_ISEXPERIMENTAL, False) or
+                is_preview_from_semantic_version(item["metadata"]['version']))
```

Comment:
```suggestion
    return item["metadata"].get(EXT_METADATA_ISPREVIEW, False) and
                item["metadata"].get(EXT_METADATA_ISEXPERIMENTAL, False) and
                not is_preview_from_semantic_version(item["metadata"]['version']
```
May I ask could the above code be expressed more concisely?

Reply:
Also, I'd like to remind that some old extensions may have already been GA to a stable version, but they may have forgotten to change the `azext.isPreview`.
Maybe we can either correct them or simply ignore the `azext.isPreview` if it doesn't have any practical effect

Reply:
does `item` always have value and contains `metadata` key?

Reply:
> does `item` always have value and always contain `metadata` key?

Yes, it's a property defined in class Extension and initially is an empty dict.

Reply:
> In addition, I'd like to remind that some old extensions may have already been GA to a stable version, but they may have forgotten to change the `azext.isPreview`. Maybe we can either correct them, or simply ignore the `azext.isPreview` if it doesn't have any practical effect

yes. For extension packages, we plan to correct their package versions and version tags in future pipelines.